slash/parallel/server.py

```diff
@@ -56,7 +56,8 @@ def __init__(self, tests):
         for i in range(len(tests)):
             self.unstarted_tests.put(i)
         self.clients_last_communication_time = {}
-        self.collection = [[test.__slash__.file_path, test.__slash__.function_name, test.__slash__.variation.id] for test in self.tests]
+        self.collection = [[test.__slash__.file_path, test.__slash__.function_name, test.__slash__.variation.safe_repr] for test in self.tests]
```

Comment:
`safe_repr` is lossy - it is not guaranteed to sort reliably. The numeric ids must be preserved. If the problem is comaprison, you can use frozenset or some other variant computed off the safe_repr...